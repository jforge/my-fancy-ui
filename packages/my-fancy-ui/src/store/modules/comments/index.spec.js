// @flow

import {
  sagas,
  selectors,
  actions,
  actionTypes,
  initialState,
  reducer
} from './index.js';

describe('exports', () => {
  it('should export a property of key "actionTypes".', () => {
    expect(actionTypes).toBeDefined();
  });

  it('should export a property of key "actions".', () => {
    expect(actions).toBeDefined();
  });

  it('should export a property of key "selectors".', () => {
    expect(selectors).toBeDefined();
  });

  it('should export a property of key "sagas".', () => {
    expect(sagas).toBeDefined();
  });
});

describe('initialState', () => {
  it('should be an object containing the default state structure.', () => {
    expect(typeof initialState).toBe('object');
  });
});

describe('reducer()', () => {
  it('should be a function.', () => {
    expect(typeof reducer).toBe('function');
  });

  it('reducer()[actionTypes.SET_COMMENTS] should transform and set the api response into the "byId" state structure', () => {
    const initialState = {
      commentsById: {}
    };
    // $FlowFixMe: suppressing this error since it is a test case
    const action = actions.setComments([
      {id: 'foo', name: 'Foo'},
      {id: 'bar', name: 'Bar'}
    ]);
    const result = reducer(initialState, action);

    expect(result).toMatchSnapshot();
  });
});
