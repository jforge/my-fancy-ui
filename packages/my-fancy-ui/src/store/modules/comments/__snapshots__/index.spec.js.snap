// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer() reducer()[actionTypes.SET_COMMENTS] should transform and set the api response into the "byId" state structure 1`] = `
Object {
  "commentsById": Object {
    "bar": Object {
      "id": "bar",
      "name": "Bar",
    },
    "foo": Object {
      "id": "foo",
      "name": "Foo",
    },
  },
}
`;
